spring-boot-autoconfigure/src/main/java/org/axonframework/boot/autoconfig/KafkaAutoConfiguration.java

```diff
@@ -82,29 +82,29 @@ public KafkaAutoConfiguration(KafkaProperties properties) {
     @ConditionalOnMissingBean
     @Bean(initMethod = "start", destroyMethod = "shutDown")
     @ConditionalOnBean(ProducerFactory.class)
-    public KafkaPublisher<String, byte[]> kafkaPublisher(ProducerFactory<String, byte[]> producerFactory,
+    public KafkaPublisher<String, byte[]> kafkaPublisher(ProducerFactory<String, byte[]> kafkaProducerFactory,
                                                          EventBus eventBus,
-                                                         KafkaMessageConverter<String, byte[]> messageConverter,
+                                                         KafkaMessageConverter<String, byte[]> kafkaMessageConverter,
                                                          AxonConfiguration configuration) {
         return new KafkaPublisher<>(KafkaPublisherConfiguration.<String, byte[]>builder()
                                             .withTopic(properties.getDefaultTopic())
-                                            .withMessageConverter(messageConverter)
-                                            .withProducerFactory(producerFactory)
+                                            .withMessageConverter(kafkaMessageConverter)
+                                            .withProducerFactory(kafkaProducerFactory)
                                             .withMessageSource(eventBus)
                                             .withMessageMonitor(configuration
                                                                         .messageMonitor(KafkaPublisher.class, "kafkaPublisher"))
                                             .build());
     }
 
     @ConditionalOnMissingBean
+    @ConditionalOnBean(ConsumerFactory.class)
```

Comment:
I think it should also be conditional on the availability of KafkaMessageConverter. Otherwise, adding Kafka dependencies but not Axon-Kafka will result in a failure.